spring-cloud-sleuth-core/src/main/java/org/springframework/cloud/sleuth/instrument/web/ZipkinHttpSpanInjector.java

```diff
@@ -0,0 +1,53 @@
+package org.springframework.cloud.sleuth.instrument.web;
+
+import java.util.Map;
+
+import org.springframework.cloud.sleuth.HttpSpanInjector;
+import org.springframework.cloud.sleuth.Span;
+import org.springframework.cloud.sleuth.SpanTextMap;
+import org.springframework.util.StringUtils;
+
+/**
+ * Default implementation of {@link HttpSpanInjector}, compatible with Zipkin propagation.
+ *
+ * @author Marcin Grzejszczak
+ * @since 1.2.0
+ */
+public class ZipkinHttpSpanInjector implements HttpSpanInjector {
+
+	@Override
+	public void inject(Span span, SpanTextMap carrier) {
+		setIdHeader(carrier, Span.TRACE_ID_NAME, span.getTraceId());
```

Comment:
why wouldn't you do a loop or switch here instead of changing to a map? seems similar amount of code or possibly less